### PR TITLE
[DEVELOPMENT|AXE-CORE] Do not show empty review headers

### PR DESCRIPTION
**Pull request by Mottie**
## Description

This PR ensures that review headers aren't empty for `ArrayFields` on the review page
- A set `reviewTitle` will now take precedence over the ui schema and page title.
- When no ui schema title is available, or a string that is empty of contains only whitespace, it ignores this entry and falls back to the page title

Previously, the ui schema title would have priority over the review title, and it would not be ignored if it contained whitespace only.

Related: https://github.com/department-of-veterans-affairs/va.gov-team/issues/12672

## Testing done

Added unit test: `reviewTitle` > `uiSchema['ui:title']` > `pageTitle`

## Screenshots

<details><summary>Periods of Confinement header now shows</summary>

Screen Shot 2020-09-02 at 3 04 37 PM</details>

## Acceptance criteria
- [x] Shows review title first, then ui schema title (if non-empty/non-whitespace filled), then page title

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ndsprinkle — APPROVED** [review]

File: src/platform/forms-system/src/js/review/ArrayField.jsx
Comment (cvalarida) [inline src/platform/forms-system/src/js/review/ArrayField.jsx:213]: This added class displays an `h3` as an `h4`. Has this passed design review?

cc @CrystabelReiter

File: src/platform/forms-system/test/js/review/ArrayField.unit.spec.jsx
Comment (cvalarida) [inline src/platform/forms-system/test/js/review/ArrayField.unit.spec.jsx:398]:
```suggestion
        pageTitle="Page Title"
```
Just in case.

**Review by cvalarida — COMMENTED** [review]
Code and tests look great. Nice and simple.

I just want to make sure the design has been signed off before giving it the approval.

**Comment by cvalarida** [comment]
Before (visual heading level 3):
image

After (visual heading level 4):
image

Does this look good to you, @CrystabelReiter?

**Comment by CrystabelReiter** [comment]
@cvalarida Looks good to me

**Review by cvalarida — APPROVED** [review]

**Pull request merged**
